### PR TITLE
Convert IncomingCallBox to a toast

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/17912

<hr>

Screenshot_20210724_210033
Screenshot_20210724_205938

<hr>

element-web notes: Improve and move the incoming call toast

### CODE REVIEW

File: src/toasts/IncomingCallToast.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+export default class IncomingCallToast extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            silenced: false,
+        };
+    }
+
+    componentDidMount = () => {
```
Comment (gsouquet) [inline src/toasts/IncomingCallToast.tsx:52]:
```suggestion
    public componentDidMount = (): void => {
```

File: src/components/structures/ToastContainer.tsx [hunk 2]
```diff
@@ -73,16 +73,22 @@ export default class ToastContainer extends React.Component<{}, IState> {
...
+
+            toast = supplyWholeBody
+                ? content
+                : <div className={toastClasses}>
+                    <div className="mx_Toast_title">
+                        <h2>{ title }</h2>
+                        <span>{ countIndicator }</span>
+                    </div>
+                    <div className="mx_Toast_body">{ content }</div>
+                </div>;
```
Comment (gsouquet) [inline src/components/structures/ToastContainer.tsx:87]: The `supplyWholeBody` prop appear a bit odd to me.
Now that the `title` is optional wouldn't it work to conditionally display `mx_Toast_title` based on whether the prop exists or not and keep `mx_Toast_body` always

I'm probably the obvious here

Comment (SimonBrandner) [reply src/components/structures/ToastContainer.tsx:87]: This is of course a possibility but the problem is that we need to style the whole thing, so we would need to pass in another `className` for the container and maybe the toast itself. So this just seemed simpler to do but it's not exactly nice, though I think the other thing wouldn't be either.

Perhaps there is a better way to reorganize all of this to make a bit simpler :thinking:

Comment (SimonBrandner) [reply src/components/structures/ToastContainer.tsx:87]: Yeah, passing in another className seems to be clearer

**Review by gaelledel — APPROVED** [review]
Great! Thank you for the changes :)

**Review by gsouquet — APPROVED** [review]

**Pull request merged**